src/ActionBar/ActionBar.tsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
```

Comment:
For this and all others, it looks like you `git add [filename].tsx` files and then did `git rm [filename].js` file. This will result in a loss of source control history on the `[filename].js`.

Please un delete the `*.js` files, then perform a `git mv`, such as `git mv src/ActionBar/ActionBar.js src/ActionBar/ActionBar.tsx`

This will allow git associate the file history.

Reply:
@bcullman - Do you happen to know the git command to un delete the *.js files? I found a command git checkout remotes/origin/develop -- src/ActionBar/ActionBar.js but didn't know if there is an easier way to retrieve the deleted files.

Reply:
i believe that `git checkout HEAD -- path/to/file/file.cc` will work

Reply:
I was able to get the file back by doing, git checkout remotes/origin/develop -- src/ActionBar/ActionBar.js, but when i renamed ActionBar.js to ActionBar.tsx it still shows the .js file as deleted. Is this correct behavior because the history will show it a being renamed?

Reply:
`git log --follow src/ActionBar/ActionBar.tsx` will follow the commit history to the file back from before the rename, but only after you commit, and only if you make no other edits to the file

```
git log --follow src/ActionBar.js
git mv src/ActionBar/ActionBar.js src/ActionBar/ActionBar.tsx
git add src/ActionBar/ActionBar.tsx
git commit -m "rename ActionBar.js -> ActionBar.tsx"
git log --follow src/ActionBar.tsx
...
[further edits to the ActionBar.tsx file]
```

Reply:
Ok i made the changes to the ActionBar, Alert, Badge and Breadcrumb files. Does it look correct? Want to make sure my steps are correct before i do all the other files.

Reply:
looks like only ActionBar retained history (see ` git log --follow ActionBar.tsx` vs `git log --follow Badge.tsx`), and even then, the history is now exceptionally noisy. 

Not sure if this repo forces squash commits or not, but if not, this will be messy history.

Honestly, it might be better to copy paste the directory elsewhere to retain your typescript changes. Then, destroy the branch, and then create a fresh branch where the first thing that is done is a proper git mv for files.

Separately, consider my other comment on this branch, with respect to linting.

Reply:
Thanks, yes I do see the history getting crazy. The best solution would probably be to create a new branch if we decide to go with TypeScript.

I have read your comments and do agree that maybe TypeScript is not the best solution right now. I will be posting a question to the team to get their feelings on if TypeScript should be implemented yet or try to align with SAP Concur with the linting rules for the components

Reply:
@bcullman - Can you take a look at this branch (https://github.com/SAP/fundamental-react/tree/typescript-support). First step I did was rename all .js file to .tsx, push those changes to the branch. Then updated all the code to be the TypeScript version and pushed to branch.

Does the Git History look good?